### ISSUE TITLE
[coc.nvim] Server languageserver.vimls failed to start: Launching server "languageserver.vimls" using command vim-language-server failed.

### ISSUE THREAD

**Issue by Linerre** [iamcco/vim-language-server#56]
**Describe the bug**
I tried `vim-langauge-server` (installed via `npm`) with both `vim-lsc` plugin and `coc.nvim` plugin, but none of them could start the server properly. 

**To Reproduce**
1. I installed `vim-language-server` with `npm install -g vim-language-server`. Below screenshots can confirm my installation is successful.
2. I copied all the config from README to my `coc-settings.json`, and there is nothing more in my coc config
3. When I tried to start vim to open my vimrc (or whichever *.vim file), I got the titled error:
4. The error message when using `vim-language-server` with `vim-lsc` is different, so I reported the issue there [#421](https://github.com/natebosch/vim-lsc/issues/421)

**Expected behavior**
I just expected `vim-language-server` to act as what is shown in the project's README with the given/recommended config.

**Screenshots**
Here are screenshots relevant to the problem:
1. `vim-language-server` was installed successfully
![vim-language-server](https://user-images.githubusercontent.com/49512984/115330867-7517f400-a1c7-11eb-96bf-dbfe1c0d088b.png)
2. start the server from the command line; **no errors**
![start server](https://user-images.githubusercontent.com/49512984/115330947-95e04980-a1c7-11eb-8af7-b8ca63647278.png)
3. Entering a `.vim` file, say my *vimrc*, I got an error from coc.nvim:
![server failed to start](https://user-images.githubusercontent.com/49512984/115331127-ed7eb500-a1c7-11eb-8e9a-604cf4a76f36.png)
4. I run `CocDiagnostics`, but got nothing:
![coc-dianostics](https://user-images.githubusercontent.com/49512984/115331199-1010ce00-a1c8-11eb-9c2e-97e2a699e6c0.png)
5. my `coc-settings.json` is very simple (I removed comments and turned `isNeovim` to `false`):
![coc-settings.json](https://user-images.githubusercontent.com/49512984/115331384-5ebe6800-a1c8-11eb-9a39-dc1e836115aa.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Vim 8.2 （installed via `scoop`)
 - Terminal: ConEmu 
 - Shell: PowerShell

Log:
If I start vim with `vim -V15mylog`, there will be over **11,000** lines of log. So I don't think a log file will be helpful at here. If you need me to log the error in specific way, please let me know.
```
```

Thank you.

**Comment by iamcco** [comment]
Open terminal in vim then check if command `vim-language-server` exists.

**Comment by Linerre** [comment]
Thanks for reply. Here is the result. The first attempt failed because I forgot to add `--stdio`. See the second please:
![terminal-vim](https://user-images.githubusercontent.com/49512984/115346175-6d197d80-a1e2-11eb-97b8-a87055b10943.png)

After trying to start the server from the terminal in Vim, it just hang there, as if I started it from my normal terminal emulator.
What should I do next?

**Comment by iamcco** [comment]
Paste coc log here.

**Comment by iamcco** [comment]
add the `"trace.server": "verbose",` in `vimls` config and restart vim and open vim file. then `:CocCommand workspace.showOutput` and select `languageserver.vimls` to open the log channel.

**Comment by Linerre** [comment]
Thank you. I can follow the first two steps:
1. add `"trace.server": "verbose", ` in my `vimls` config
![trace-server](https://user-images.githubusercontent.com/49512984/115358994-cee0e400-a1f0-11eb-9ff2-18e5c7e60f83.png)
2. restart vim, open a vim file, and run the command `:CocCommand workspace.showOutput`:
![coc-command](https://user-images.githubusercontent.com/49512984/115359179-0059af80-a1f1-11eb-9bd1-667fc4f15901.png)

But I have no idea how to do the last one. After running cmd in step 2, I couldn't seem to select anything from the buffer. Would you care to give me a few more hints? Many thanks.

**Comment by Linerre** [comment]
> Edit: With the coc log we just can see serve start fail but don't know why. So to check the server log channel to see if there are something useful.

The wired part is that the server is indeed working. See the screenshots below. I took them immediately after I replied to your last comment, without doing anything to vim (no restarting, no re-configuring).  The auto-completion works very well as expected: suggesting me with built-in viml keywords and options:
![server1](https://user-images.githubusercontent.com/49512984/115360055-d785ea00-a1f1-11eb-9966-2af769be690b.png)
![server2](https://user-images.githubusercontent.com/49512984/115360064-d9e84400-a1f1-11eb-960e-b30f769eceb7.png)

I don't know if this is a Windows-only problem. I hope it is not. Again, thank you for your reply. I really appreciate it.

**Comment by iamcco** [comment]
Have you install coc-vimlsp extension? check with `:CocList extensions`. if yes then you don't need to install vim-language-server and those config.

**Comment by iamcco** [comment]
> Thank you. I can follow the first two steps:
> 
> 1. add `"trace.server": "verbose", ` in my `vimls` config
>    ![trace-server](https://user-images.githubusercontent.com/49512984/115358994-cee0e400-a1f0-11eb-9ff2-18e5c7e60f83.png)
> 2. restart vim, open a vim file, and run the command `:CocCommand workspace.showOutput`:
>    ![coc-command](https://user-images.githubusercontent.com/49512984/115359179-0059af80-a1f1-11eb-9bd1-667fc4f15901.png)
> 
> But I have no idea how to do the last one. After running cmd in step 2, I couldn't seem to select anything from the buffer. Would you care to give me a few more hints? Many thanks.

There is only one channel so you don't need to select. Seems like there is no `vim-language-server` command when coc start the server. change the command to use absolute path like `/path/to/vim-language-server` and test again.

**Comment by Linerre** [comment]
OMG! That's interesting.  Yes, I installed it. My bad. I mistakenly thought I should install both, one server and one extension for the server. I removed the server config in `coc-settings.json` but keep the `coc-vimlsp` extension. Now the error has gone and the LSP is functioning without any problems.
![coc-vimlsp](https://user-images.githubusercontent.com/49512984/115372534-972c6900-a1fd-11eb-9eee-241968985fe3.png)
![coc-vimlsp-working](https://user-images.githubusercontent.com/49512984/115372773-d22e9c80-a1fd-11eb-8b37-c36b45927a15.png)

I think I'll keep the server installed via npm, since other LSP plugins will rely on it. Many thanks for your time and help.
